src/engine/QueryPlanner.cpp

```diff
@@ -790,245 +840,476 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::seedWithScansAndText(
+  std::cout << "Turned " << triple.asString() << " into " << std::endl;
+  std::cout << out.str() << std::endl << std::endl;
+  pattern.recomputeIds();
+  return optimize(&pattern);
+}
+
+ParsedQuery::GraphPattern QueryPlanner::seedFromPropertyPath(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  switch (path._operation) {
+    case PropertyPath::Operation::ALTERNATIVE:
+      return seedFromAlternative(left, path, right);
+    case PropertyPath::Operation::INVERSE:
+      return seedFromInverse(left, path, right);
+    case PropertyPath::Operation::IRI:
+      return seedFromIri(left, path, right);
+    case PropertyPath::Operation::SEQUENCE:
+      return seedFromSequence(left, path, right);
+    case PropertyPath::Operation::TRANSITIVE:
+      return seedFromTransitive(left, path, right);
+    case PropertyPath::Operation::TRANSITIVE_MAX:
+      return seedFromTransitiveMax(left, path, right);
+    case PropertyPath::Operation::TRANSITIVE_MIN:
+      return seedFromTransitiveMin(left, path, right);
+  }
+  AD_THROW(
+      ad_semsearch::Exception::NOT_YET_IMPLEMENTED,
+      "No implementation for creating a seed from a property path of type " +
+          ((int)path._operation));
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromSequence(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  if (path._children.size() == 0) {
+    AD_THROW(ad_semsearch::Exception::BAD_INPUT,
+             "Tried processing a sequence property path node without any "
+             "children.");
+  } else if (path._children.size() == 1) {
+    LOG(WARN) << "Processing a sequence property path that has only one child."
+              << std::endl;
+    return seedFromPropertyPath(left, path, right);
+  }
+  std::vector<ParsedQuery::GraphPattern> childPlans;
+  childPlans.reserve(path._children.size());
+
+  std::vector<std::string> tmpVarNames;
+  for (size_t i = 0; i < path._children.size(); i++) {
+    if (i == 0) {
+      tmpVarNames.push_back(generateRandomVarName());
+      childPlans.push_back(
+          seedFromPropertyPath(left, path._children[i], tmpVarNames.back()));
+    } else if (i + 1 == path._children.size()) {
+      childPlans.push_back(
+          seedFromPropertyPath(tmpVarNames.back(), path._children[i], right));
+    } else {
+      tmpVarNames.push_back(generateRandomVarName());
+      childPlans.push_back(
+          seedFromPropertyPath(tmpVarNames[tmpVarNames.size() - 2],
+                               path._children[i], tmpVarNames.back()));
+    }
+  }
+
+  // Merge all the child plans into one graph pattern
+  ParsedQuery::GraphPattern p;
+  for (ParsedQuery::GraphPattern& child : childPlans) {
+    p._children.insert(p._children.end(), child._children.begin(),
+                       child._children.end());
+    // Ensure the child does not try to delete the pointers
```

Comment:
Maybe we should really change these child pointers to `std::shared_ptr` but I realize this is a large change, you could add a TODO

Reply:
Seems like a good idea to me. I've added a TODO for now.